cloudcix_primitives/storage_kvm.py

```diff
@@ -55,38 +59,48 @@ def build(
     # Define message
     messages = {
         1000: f'1000: Successfully created storage {storage}',
-        3021: f'3021: Failed to connect to the host {host}',
-        3022: f'3022: Failed to create storage_kvm {domain_path}{storage} on the host {host}'
+        1001: f'1001: Already storage {storage} file exists at {domain_path}',
+        3021: f'3021: Failed to connect to the host {host} for the payload read_storage_file',
+        3022: f'3022: Failed to connect to the host {host} for the payload create_storage_file',
+        3023: f'3023: Failed to create storage_kvm {domain_path}{storage} on the host {host}',
     }
 
-    message_list = {}
-
-    # Define payload
-    payload = f'qemu-img create -f qcow2 {domain_path}{storage} {size}G'
-
-    # Create storage using SSH communication
-    response = comms_ssh(
-        host_ip=host,
-        payload=payload,
-        username='robot',
-    )
-    if response['channel_code'] != CHANNEL_SUCCESS:
-        msg = f'{messages[3021]}\n ' \
-                f'channel_code: {response["channel_code"]}s.\n' \
-                f'channel_message: {response["channel_message"]}\n' \
-                f'channel_error: {response["channel_error"]}'
-        message_list.append(msg)
-        success = False
-    if response['payload_code'] != SUCCESS_CODE:
-        msg = f'{messages[3022]}\n ' \
-                f'payload_code: {response["payload_code"]}s.\n' \
-                f'payload_message: {response["payload_message"]}\n' \
-                f'payload_error: {response["payload_error"]}'
-        message_list.append(msg)
-        success = False
-
-    message_list.append(messages[1000])
-    return success, message_list
+    def run_host(host, prefix, successful_payloads):
+        rcc = SSHCommsWrapper(comms_ssh, host, 'robot')
+        fmt = HostErrorFormatter(
+            host,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads,
+        )
+
+        payloads = {
+            'read_storage_file': f'qemu-img info {domain_path}{storage}',
+            'create_storage_file': f'qemu-img create -f qcow2 {domain_path}{storage} {size}G',
+        }
+
+        ret = rcc.run(payloads['read_storage_file'])
+        if ret["channel_code"] != CHANNEL_SUCCESS:
+            return False, fmt.channel_error(ret, messages[prefix + 1]), fmt.successful_payloads
+        if ret["payload_code"] == SUCCESS_CODE:
```

Comment:
Can you follow the pattern outlined in [ns.py](https://github.com/CloudCIX/primitives/blob/284bccf2006186a2fe7a6c3a2e46b8ceb1ce3349/cloudcix_primitives/ns.py#L144) which determines it the namespace already exists?

Also we had a discussion this morning regarding this scenario:

1. It exists and is the same size as in request. - return True, build completed
2. It exists and is a different size than in request - return False, different request sent to build the same storage drive with different data - there is a problem.

Reply:
done

Reply:
```python
        create_storage = True
        if ret["payload_code"] == SUCCESS_CODE:
            # No need to create storage drive exists already
            create_storage = False
            # Extract the size of the file that exists and return error if it does not correspond with request.
            size_match = re.search(r'virtual size: (\S+)', ret["payload_message"].strip())
            checked_size = size_match.group(1) if size_match else 0
            if int(checked_size) != int(size):
                 return False, messages[3001], fmt.successful_payloads
        fmt.add_successful('read_storage_file', ret)

        if create_storage:
             # Payload request to create storage goes here
```

This is what I was referring to when I referenced the ns primitive.

Reply:
Also `int(None)` will raise a TypeError which is why I changed `None` to `0` in the above.

Reply:
When file already exists say 20GB and requesting to build 20GB, then instead of returning True as same size file exists, your code is recreating it again instead of returning True and message=' already same sized file exists'

We should avoid recreating the file, because chances are there if build is called on storage file already used by a VM then recreating will remove the VMs.

So that's why I was doing if(int(checked_size) != int(size)) and else.

Reply:
It is not because I have set the variable `create_storage` to False if the `payload_code` in the response from the rcc was successful.

Reply:
Then the `if create_storage is True:` (appended is True to above), create the storage drive. 

Errors have been returned if 
1. `channel_code` in response from rcc was not successful
2. Drive exists but it is a different size to that in the request

Reply:
okay got it

Reply:
Done